crates/ruff/src/rules/pylint/rules/duplicate_bases.rs

```diff
@@ -0,0 +1,42 @@
+use std::hash::BuildHasherDefault;
+
+use rustc_hash::FxHashSet;
+use rustpython_parser::ast::{self, Expr, ExprKind, Identifier};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct DuplicateBases {
+    base: String,
+    class: String,
+}
+
+impl Violation for DuplicateBases {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DuplicateBases { base, class } = self;
+        format!("Duplicate base `{base}` for class `{class}`")
+    }
+}
+
+/// PLE0241
+pub(crate) fn duplicate_bases(checker: &mut Checker, name: &str, bases: &[Expr]) {
+    let mut seen: FxHashSet<&Identifier> =
+        FxHashSet::with_capacity_and_hasher(bases.len(), BuildHasherDefault::default());
+    for base in bases {
+        if let ExprKind::Name(ast::ExprName { id, .. }) = &base.node {
+            if !seen.insert(id) {
+                checker.diagnostics.push(Diagnostic::new(
+                    DuplicateBases {
+                        base: id.to_string(),
+                        class: name.to_string(),
+                    },
+                    base.range(),
+                ));
+            }
+        }
```

Comment:
```suggestion
        if let Some(name) = base.node.expr_name() {
            if !seen.insert(name.id) {
                checker.diagnostics.push(Diagnostic::new(
                    DuplicateBases {
                        base: id.to_string(),
                        class: name.to_string(),
                    },
                    base.range(),
                ));
            }
        }
```

Since this is going to be used as an example, I leave a possible other form of expr matching.